### PR TITLE
SQL DataBase

### PR DESCRIPTION
**Pull request by YevhenDov**

### CODE REVIEW

File: DataBase/src/main/java/com/development/JdbcUtil.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.development;
+
+public class JdbcUtil {
+    public static void main(String[] args) {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/JdbcUtil.java]: why do you need it here?

Comment (YevhenDov) [reply DataBase/src/main/java/com/development/JdbcUtil.java]: it not end. I'm working

File: DataBase/src/main/java/com/development/service/CRUD.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+public class CRUD {
+    private static final Logger LOGGER = Logger.getLogger(CRUD.class);
+
+    private List<Skill> skills = new ArrayList<>();
+    private List<Company> companies = new ArrayList<>();
+    private List<Project> projects = new ArrayList<>();
+    private List<Customer> customers = new ArrayList<>();
+    private List<Developer> developers = new ArrayList<>();
+
+    public void createData(String tableName, String values) {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/service/CRUD.java]: too complicated logic
create a crud repository per table

File: DataBase/src/main/java/com/development/model/Customer.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.development.model;
+
+import lombok.AllArgsConstructor;
+
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/model/Customer.java]: better add annotation data or just annotation getter

File: DataBase/src/main/java/com/development/service/Connector.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class Connector {
+    private static final Logger LOGGER = Logger.getLogger(Connector.class);
+
+    private static final String USER = "root";
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/service/Connector.java]: bettre move it yo the property file

File: DataBase/src/main/java/com/development/service/CRUD.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            }
+        } catch (SQLException e) {
+            LOGGER.info(e.getMessage());
+        }
+    }
+
+    public void updateData(String tableName, String column, String value, String condition) {
+        String request = "UPDATE " + tableName + " SET " + column + " = " + value + " WHERE " + condition;
+
+        Connection connection = Connector.getConnection();
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/service/CRUD.java]: dont forget to close connection

File: DataBase/src/main/java/com/development/service/CRUD.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    private List<Company> companies = new ArrayList<>();
+    private List<Project> projects = new ArrayList<>();
+    private List<Customer> customers = new ArrayList<>();
+    private List<Developer> developers = new ArrayList<>();
+
+    public void createData(String tableName, String values) {
+        String request = null;
+        switch (tableName) {
+            case "developers": {
+                request = "INSERT INTO developers(developer_name, age, gender) VALUES (" + values + ")";
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/service/CRUD.java]: better use constants for queries

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompanyDao {
+    private static final Logger LOGGER = Logger.getLogger(CRUD.class);
+    private List<Company> companies = new ArrayList<>();
+
+    public void createData(String values) {
+        String request;
+        request = "INSERT INTO companies(developer_name, age, gender) VALUES (" + values + ")";
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: please resolve all comments
use for queries constants

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+        Execute(request);
+    }
+
+    public void readData() {
+        ResultSet resultSet;
+        String request = "SELECT * FROM developers";
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: the same

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        Execute(request);
+    }
+
+    public void deleteData(String condition) {
+        String request = "DELETE FROM companies WHERE " + condition;
+
+        Execute(request);
+    }
+
+    private void Execute(String request) {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: use lowerCamelCase for method names

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        Connection connection = Connector.getConnection();
+        try (PreparedStatement statement = connection.prepareStatement(request)) {
+            statement.executeUpdate();
+            connection.close();
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better close connection in finally block

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompanyDao {
+    private static final Logger LOGGER = Logger.getLogger(CompanyDao.class);
+    private List<Company> companies = new ArrayList<>();
+
+    public void createData(String values) {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better name insert and pass as parameter here Company company

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    public void updateData(String column, String value, String condition) {
+        String request = "UPDATE companies SET " + column + " = " + value + " WHERE " + condition;
+
+        Execute(request);
+    }
+
+    public void deleteData(String condition) {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better delete by id
so pass as param id

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    private List<Company> companies = new ArrayList<>();
+
+    public void createData(String values) {
+        String request;
+        request = "INSERT INTO companies(developer_name, age, gender) VALUES (" + values + ")";
+
+        Execute(request);
+    }
+
+    public void readData() {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: why it returns nothing?

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    public void createData(Company company, String values) {
+        final String REQUEST ="INSERT INTO companies("
+                + company.getId() + ","
+                + company.getName() + ","
+                + company.getAddress() + ") VALUES (" + values + ")";
+
+        execute(REQUEST);
+    }
+
+    public List<Company> readData() throws SQLException {
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better name getAll() or findAll()

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompanyDao {
+    private static final Logger LOGGER = Logger.getLogger(CompanyDao.class);
+    private List<Company> companies = new ArrayList<>();
+
+    public void createData(Company company, String values) {
+        final String REQUEST ="INSERT INTO companies("
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better make it constants as field of the class

File: DataBase/src/main/java/com/development/dao/CompanyDao.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.List;
+
+public class CompanyDao {
+    private static final Logger LOGGER = Logger.getLogger(CompanyDao.class);
+    private List<Company> companies = new ArrayList<>();
+
+    private final String CREATE_REQUEST = "INSERT INTO companies(company_name, address) VALUES (?, ?)";
+    private final String READ_REQUEST = "SELECT * FROM companies_project";
+    private final String UPDATE_REQUEST = "UPDATE companies SET company_name = 'Google' WHERE company_name = ?";
+    private final String DELETE_REQUEST = "DELETE FROM companies WHERE company_name = ?";
```
Comment (Taniann) [inline DataBase/src/main/java/com/development/dao/CompanyDao.java]: better delete by id

**Pull request merged**